### PR TITLE
Add outputs, add clients_custom_data var, allow extending ec2 policies

### PR DESCRIPTION
**Pull request by kristina-solovyova**

### CODE REVIEW

File: modules/clients/main.tf [hunk 1]
```diff
@@ -113,7 +113,13 @@ resource "aws_launch_template" "this" {
       })
     }
   }
-  user_data  = local.custom_data
+  user_data = local.custom_data
+  lifecycle {
+    precondition {
+      condition     = var.alb_dns_name == null ? var.weka_cluster_size > 0 : true
+      error_message = "Please povide weka_cluster size in case of not using ALB"
```
Comment (tigrawap) [inline modules/clients/main.tf:120]: this is weird requirement, we probably can get info we need here in a different way...i understand this PR does not add it, but just validates

**Review by tigrawap — APPROVED** [review]

File: modules/clients/main.tf (hunk 1 above)
Comment (tigrawap) [reply modules/clients/main.tf]: why it needs to know the size? (did not go to see how variable is used..)

**Comment by tigrawap** [comment]
> @tigrawap @assafgi I removed `weka_cluster_size` var from both clients and protocols modules checked with asg min_size usage, it works fine
> 
> main.tf:
> 
> ```
> module "weka_deployment" {
>   source             = "../../"
>   prefix             = "ks"
>   cluster_name       = "test"
>   availability_zones = ["eu-west-1a"]
>   allow_ssh_cidrs   = ["0.0.0.0/0"]
>   get_weka_io_token = var.get_weka_io_token
>   additional_instance_iam_policy_statement = [{
>     Action   = ["s3:ListBucket"]
>     Effect   = "Allow"
>     Resource = ["*"]
>   }]
>   clients_number = 1
>   create_alb = false
> }
> ```
> 
> from client's logs:
> 
> ```
> + get_private_ips
> ++ aws autoscaling describe-auto-scaling-groups --auto-scaling-group-name ks-test-autoscaling-group --query 'AutoScalingGroups[].Instances[].InstanceId' --output text --region eu-west-1
> + instance_ids='i-03dd40790d9618f5b     i-05c7daa00257b81fe     i-0630e64608887ebd8     i-0bc45c54b4f12cf6e     i-0bfee3c5280c040eb     i-0c035988d6f901d44'
> ++ aws autoscaling describe-auto-scaling-groups --auto-scaling-group-name ks-test-autoscaling-group --query 'AutoScalingGroups[].MinSize' --output text --region eu-west-1
> + cluster_min_size=6
> ++ aws ec2 describe-instances --instance-ids i-03dd40790d9618f5b i-05c7daa00257b81fe i-0630e64608887ebd8 i-0bc45c54b4f12cf6e i-0bfee3c5280c040eb i-0c035988d6f901d44 --query 'Reservations[].Instances[
> ].PrivateIpAddress' --output text --region eu-west-1
> + private_ips='10.0.1.62        10.0.1.207      10.0.1.167      10.0.1.7        10.0.1.123      10.0.1.204'
> + private_ips_array=($private_ips)
> + length=6
> + expected_length=6
> + '[' 6 -ge 6 ']'
> + break
> + ips=("${private_ips_array[@]}")
> + backend_ip=10.0.1.62
> + retry_weka_install
> + retry_max=60
> + retry_sleep=30
> + count=60
> + '[' 60 -gt 0 ']'
> + curl --fail -o install_script.sh 10.0.1.62:14000/dist/v1/install
> ...
> + echo 'Tue Dec 19 17:07:58 UTC 2023: wekafs mount complete'
> Tue Dec 19 17:07:58 UTC 2023: wekafs mount complete
> + rm -rf /tmp/weka
> ++ date -u
> + echo 'Tue Dec 19 17:07:58 UTC 2023: client setup complete'
> Tue Dec 19 17:07:58 UTC 2023: client setup complete
> ```

If it was required param before change - need to convert it to optional param, with no meaning and deprecation notice
otherwise it will break existing deployments, i.e they will fail to upgrade

File: modules/clients/variables.tf [hunk 2]
```diff
@@ -3,9 +3,10 @@ variable "subnet_id" {
   description = "Id of the subnet"
 }
 
-variable "weka_cluster_size" {
+variable "weka_cluster_size" { # tflint-ignore: terraform_unused_declarations
```
Comment (assafgi) [inline modules/clients/variables.tf:6]: why not removing it? for customers who might use it? I think it is OK to remove

**Review by assafgi — APPROVED** [review]

**Pull request merged**